### PR TITLE
Adress Public API review comments

### PR DESCRIPTION
**Pull request by hulumane**
This addresses a bunch of comments that came out of the Public API review process

### CODE REVIEW

File: test/TestApps/ToastNotificationsTestApp/main.cpp [hunk 1]
```diff
@@ -659,6 +781,13 @@ std::map<std::string, bool(*)()> const& GetSwitchMapping()
         { "VerifyFailedUpdateNotificationDataWithoutPostToast", &VerifyFailedUpdateNotificationDataWithoutPostToast },
         { "VerifyUpdateToastProgressDataUsingEmptyTagAndValidGroup", &VerifyUpdateToastProgressDataUsingEmptyTagAndValidGroup },
         { "VerifyUpdateToastProgressDataUsingEmptyTagAndEmptyGroup", &VerifyUpdateToastProgressDataUsingEmptyTagAndEmptyGroup },
+        { "VerifyExplicitAppId_Unpackaged", &VerifyExplicitAppId_Unpackaged },
+        { "VerifyUnregisterAll_Unpackaged", &VerifyUnregisterAll_Unpackaged },
+        { "VerifyUnregisterAll", &VerifyUnregisterAll },
+        { "VerifyUnregisterTwice", &VerifyUnregisterTwice },
+        { "VerifyUnregisterTwice_Unpackaged", &VerifyUnregisterTwice_Unpackaged },
+        { "VerifyToastProgressDataSequence0Fail", &VerifyToastProgressDataSequence0Fail },
+        { "VerifyToastUpdateZeroSequenceFail", &VerifyToastUpdateZeroSequenceFail_Unpackaged },
```
Comment (loneursid) [inline test/TestApps/ToastNotificationsTestApp/main.cpp:790]: Seems to be a packaged / unpackaged mismatch

Comment (hulumane) [reply test/TestApps/ToastNotificationsTestApp/main.cpp:790]: Yeah im fixing this bug. New PR.

File: dev/AppNotifications/AppNotificationManager.cpp [hunk 2]
```diff
@@ -78,45 +67,67 @@ namespace winrt::Microsoft::Windows::AppNotifications::implementation
...
+                winrt::make<AppNotificationActivationCallbackFactory>().get(),
+                CLSCTX_LOCAL_SERVER,
+                REGCLS_MULTIPLEUSE,
+                &m_notificationComActivatorRegistration));
+        }
 
         GetWaitHandleForArgs().create();
     }
 
+    void AppNotificationManager::Register()
```
Comment (sharath2727) [inline dev/AppNotifications/AppNotificationManager.cpp:84]: Please don't create this, I am modifying the existing one in my PR. That should suffice. Thanks.

File: test/TestApps/ToastNotificationsTestApp/main.cpp [hunk 3]
```diff
@@ -525,19 +605,48 @@ bool VerifyUpdateToastProgressDataUsingValidTagAndEmptyGroup_Unpackaged()
...
+        toastNotificationManager.Register(activationInfo);
+
+        auto scope_exit = wil::scope_exit(
+            [&] {
+                toastNotificationManager.Unregister();
+            });
+
+        PostToastHelper(L"Tag", L"");
+
+        winrt::AppNotificationProgressData progressData = GetToastProgressData(L"SomeStatus", L"SomeTitle", 0.14, L"14%", false, 1);
```
Comment (sharath2727) [inline test/TestApps/ToastNotificationsTestApp/main.cpp:632]: sequenceNumber is set to 1 here? It won't throw right?

File: dev/AppNotifications/AppNotificationManager.cpp (hunk 2 above)
Comment (hulumane) [reply dev/AppNotifications/AppNotificationManager.cpp:86]: Just rebase

File: test/TestApps/ToastNotificationsTestApp/main.cpp [hunk 4]
```diff
@@ -525,19 +605,48 @@ bool VerifyUpdateToastProgressDataUsingValidTagAndEmptyGroup_Unpackaged()
...
-    auto progressResultOperation = toastNotificationManager.UpdateProgressDataAsync(progressData, L"Tag");
+    auto progressResultOperation = toastNotificationManager.UpdateAsync(progressData, L"Tag");
     return ProgressResultOperationHelper(progressResultOperation, winrt::AppNotificationProgressResult::Succeeded);
 }
 
+bool VerifyToastUpdateZeroSequenceFail_Unpackaged()
+{
+    try
+    {
+        Sleep(30000);
```
[the remark this answers is not in the document]
Comment (hulumane) [reply test/TestApps/ToastNotificationsTestApp/main.cpp]: removed

File: test/TestApps/ToastNotificationsTestApp/main.cpp (hunk 3 above)
Comment (hulumane) [reply test/TestApps/ToastNotificationsTestApp/main.cpp]: No because the setsequence is set to false. It can be anything.

**Review by pmpurifoy — APPROVED** [review]

File: dev/AppNotifications/AppNotificationManager.h [hunk 5]
```diff
@@ -1,27 +1,49 @@
...
-        winrt::Microsoft::Windows::AppNotifications::AppNotificationSetting Enablement();
+        winrt::Windows::Foundation::IAsyncOperation<winrt::Microsoft::Windows::AppNotifications::AppNotificationProgressResult> UpdateAsync(winrt::Microsoft::Windows::AppNotifications::AppNotificationProgressData const data, hstring const tag, hstring const group);
+        winrt::Windows::Foundation::IAsyncOperation<winrt::Microsoft::Windows::AppNotifications::AppNotificationProgressResult> UpdateAsync(winrt::Microsoft::Windows::AppNotifications::AppNotificationProgressData const data, hstring const tag);
+        winrt::Microsoft::Windows::AppNotifications::AppNotificationSetting Setting();
         winrt::Windows::Foundation::IAsyncAction RemoveByIdAsync(uint32_t notificationId);
-        winrt::Windows::Foundation::IAsyncAction RemoveWithTagAsync(hstring const tag);
-        winrt::Windows::Foundation::IAsyncAction RemoveWithTagGroupAsync(hstring const tag, hstring const group);
-        winrt::Windows::Foundation::IAsyncAction RemoveGroupAsync(hstring const group);
+        winrt::Windows::Foundation::IAsyncAction RemoveByTagAsync(hstring tag);
```
Comment (loneursid) [inline dev/AppNotifications/AppNotificationManager.h:33]: Why are we losing the constness

**Review by loneursid — APPROVED** [review]
nit: const was removed from some API. It seems like it was by accident

File: dev/AppNotifications/AppNotificationManager.h (hunk 5 above)
Comment (hulumane) [reply dev/AppNotifications/AppNotificationManager.h:33]: Good catch. Will fix.

Comment (hulumane) [reply dev/AppNotifications/AppNotificationManager.h:33]: Wasn't on purpose. I promise :)

Comment (hulumane) [reply dev/AppNotifications/AppNotificationManager.h:33]: I'll send a new PR out addressing this in a bit.

File: dev/AppNotifications/AppNotificationUtility.cpp [hunk 6]
```diff
@@ -25,51 +26,61 @@ namespace ToastABI
...
+        THROW_IF_FAILED(GetCurrentProcessPath(processName));
 
-    THROW_HR_IF(status, FAILED_WIN32(status));
-    return registeredGuidBuffer;
+        std::wstring wideStringProcessName{ processName.get() };
+        // subKey: L"Software\\Classes\\AppUserModelId\\{Path to ToastNotificationsTestApp.exe}"
+        std::wstring subKey{ c_appIdentifierPath + ConvertPathToKey(wideStringProcessName) };
+
+        wil::unique_hkey hKey;
+        THROW_IF_FAILED(RegCreateKeyEx(
```
Comment (ChrisGuzak) [inline dev/AppNotifications/AppNotificationUtility.cpp:46]: `RegCreateKeyEx` returns a Win32 error code, not an HRESULT, so this is a bug. use `THROW_IF_WIN32_ERROR`

File: dev/AppNotifications/AppNotificationUtility.cpp [hunk 7]
```diff
@@ -25,51 +26,61 @@ namespace ToastABI
...
+            nullptr /* lpClass */,
+            REG_OPTION_NON_VOLATILE,
+            KEY_ALL_ACCESS,
+            nullptr /* lpSecurityAttributes */,
+            &hKey,
+            nullptr /* lpdwDisposition */));
+
+        WCHAR registeredGuidBuffer[GUID_LENGTH];
+        DWORD bufferLength = sizeof(registeredGuidBuffer);
+        HRESULT status = RegGetValueW(
```
Comment (ChrisGuzak) [inline dev/AppNotifications/AppNotificationUtility.cpp:59]: RegGetValueW returns a win32 error code, not an HRESULT. use the correct type.
also, `status` sounds like `NTSTATUS`, use `hr` or `result`

**Pull request merged**